src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -376,10 +400,12 @@ case class WorkflowActor(workflow: WorkflowDescriptor,
    * Load whatever execution statuses are stored for this workflow, regardless of whether this is a workflow being
    * restarted, or started for the first time.
    */
-  private def createStore: Future[Map[Call, ExecutionStatus.Value]] = {
+  private def createStore: Future[ExecutionStore] = {
     dataAccess.getExecutionStatuses(workflow.id) map { statuses =>
-      workflow.namespace.workflow.calls.map {call =>
-        call -> statuses.get(call.fullyQualifiedName).get}.toMap
+      // TODO: This is NOT right. For example: statuses from DB are not just for a FQN, need the scatter index.
```

Comment:
Also the `DataAccess` API will need to be reworked for this generally.  Not sure there's a ticket that captures that...